cluster/manager.go

```diff
@@ -160,17 +160,21 @@ func (c *MemberlistManager) thisNode() HTTPNode {
 	return c.members[c.nodeName]
 }
 
-func (c *MemberlistManager) MemberList() []Node {
-	return toIf(c.memberList())
+func (c *MemberlistManager) MemberList(isReady, hasData bool) []Node {
+	return toIf(c.memberList(isReady, hasData))
 }
 
-func (c *MemberlistManager) memberList() []HTTPNode {
+func (c *MemberlistManager) memberList(isReady, hasData bool) []HTTPNode {
 	c.RLock()
-	list := make([]HTTPNode, len(c.members), len(c.members))
-	i := 0
+	list := make([]HTTPNode, 0, len(c.members))
 	for _, p := range c.members {
-		list[i] = p
-		i++
+		if isReady && !p.IsReady() {
```

Comment:
i've spent at least half a minute looking at those conditions to understand how they work, would be nice to have a comment there...